share/spack/templates/container/leap-15.dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM opensuse/leap:15.3
+MAINTAINER Christian Goll <cgoll@suse.com>
+
+ENV DOCKERFILE_BASE=opensuse          \
+    DOCKERFILE_DISTRO=leap   \
+    DOCKERFILE_DISTRO_VERSION=15.3    \
+    SPACK_ROOT=/opt/spack      \
+    DEBIAN_FRONTEND=noninteractive    \
+    CURRENTLY_BUILDING_DOCKER_IMAGE=1 \
+    container=docker
+
+RUN zypper ref && \
+    zypper up -y && \
+    zypper in -y \
+    bzip2\
+    curl\
+    file\
+    gcc-c++\
+    gcc-fortran\
+    make\
+    gzip\
+    patch\
+    patchelf\
+    python3-base \
+    python3-boto3\
+    tar\
+    xz\
+&&  zypper clean
+
+# clean up manpages
+RUN	rm -rf /var/cache/zypp/*  \
+	rm -rf /usr/share/doc/packages/* \ 
+	rm -rf /usr/share/doc/manual/*
+
+# copy spack into container
+COPY bin   $SPACK_ROOT/bin
+COPY etc   $SPACK_ROOT/etc
+COPY lib   $SPACK_ROOT/lib
+COPY share $SPACK_ROOT/share
+COPY var   $SPACK_ROOT/var
+RUN mkdir -p $SPACK_ROOT/opt/spack
```

Comment:
While this is fine for the recipes in `share/spack/docker`, the Jinja2 template file needs to be able to clone Spack at arbitrary versions for reproducibility - so they have a slightly different structure. 

There's a base template called `bootstrap-base.dockerfile` that is usually extended by each specific OS. You can have a look for instance at the CUDA Dockerfile to check an example. Let me know if anything is unclear.